silq/pulses/pulse_types.py

```diff
@@ -668,6 +668,138 @@ def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
         return waveform
 
 
+class MultiSinePulse(Pulse):
+    """MultiSinusoidal pulse
+
+    Parameters:
+        name: Pulse name
+        frequency: main (local oscillator) Pulse frequency
+        frequencies: list of Pulse frequencies
+        phases: list of Pulse phases
+        amplitude: Pulse amplitude. If not set, power must be set.
+        amplitudes: list of Pulse amplitudes
+        power: Pulse power. If not set, amplitude must be set.
+        offset: amplitude offset, zero by default
+        frequency_sideband: Mixer sideband frequency (off by default).
+        sideband_mode: Sideband frequency to apply. This feature must
+            be existent in interface. Not used if not set.
+        phase_reference: What point in the the phase is with respect to.
+            Can be two modes:
+            - 'absolute': phase is with respect to `Pulse.t_start`.
+            - 'relative': phase is with respect to t=0 (phase-coherent).
+
+        **kwargs: Additional parameters of `Pulse`.
+
+    Notes:
+        Either amplitude or power must be set, depending on the instrument
+        that should output the pulse.
+    """
+    def __init__(self,
+                 name: str = None,
+                 frequency: float = None,
+                 frequencies: List[float] = None,
+                 phases: List[float] = None,
+                 amplitude: float = None,
+                 amplitudes: List[float] = None,
+                 power: float = None,
+                 offset: float = None,
+                 frequency_sideband: float = None,
+                 sideband_mode: float = None,
+                 phase_reference: str = None,
+                 **kwargs):
+
+        super().__init__(name=name, **kwargs)
+
+        self.frequency = Parameter(initial_value=frequency, unit='Hz',
```

Comment:
Is this for the carrier frequency? If so, it needs to have a different name, else it will be too confusing. I also don't know if it's a good idea to specify the carrier frequency in a pulse, as it may lead to problems later on. For instance, what if different pulses request different carrier frequencies? I feels more like a property of the microwave interface than of the pulse, but it would be good to discuss this

Reply:
This frequency is not exactly the carrier frequency. If we have only one pulse in pulse sequence, and frequency carrier choice is set to 'center', then yes - it acts as a carrier frequency. However, if we have more pulses, e.g. SinePulse for ESR1 and MultiSinePulse for ESR2, then microwave interface treats this frequency as the main frequency of the MultiSinePulse similarly to FrequencyRampPulse. In this case, depending on the carrier choice, the carrier frequency will be determined from those two ESR1 and ESR2 frequencies of the pulses. So, in short, I didn't rename on purpose, since when microwave interface is checking pulses in pulse_sequence it works with "frequency" parameter of the pulse - I just wanted to be compatible with the interface.

Reply:
Let's discuss this, I am not sure what kind of issues you have in mind

Reply:
I looked again at all of this and generally I agree - we can remove "frequency" from the MultiSine Pulse with a few adjustments to the interface min/max frequency determination, let's shortly discuss this and I will remove it.

Reply:
corrected